src/deposit.cairo

```diff
@@ -188,8 +210,8 @@ mod Deposit {
                     total_borrowed,
```

Comment:
Lol i've written my response for an hour and now it's gone because i pressed a wrong key - my fault. I will cut it short. Basically, the current borrow capacity calculation has a problem when we have a token to be borrowed that has the number of decimal places equal to or more than that of the deposited token. Let's illustrate it with a simple example.

First, we look at a case where the deposited token is ETH with 18 decimal places and the token to be borrowed is USDC, with 6 decimal places.

```
// deposited = 1.000000000000000003 ETH = 1000000000000000003
// collateral_factor = 500000000000000000000000000 (50%)
// scale = 1000000000000000000000000000
// borrow_factor = 500000000000000000000000000 (50%)
// price: 2400n * (10n**6n) (in USDC terms)
```

Then:

```
borrow capacity = 1000000000000000003n * 500000000000000000000000000n / 1000000000000000000000000000n * 500000000000000000000000000n / 1000000000000000000000000000n = 250000000000000000n
```

Well, we already lost 0.000000000000000003 ETH right? So let's try to scale the equation by `ZK_SCALE_DECIMALS`:

```
borrow capacity = 1000000000000000003n * 1000000000000000000000000000n * 500000000000000000000000000n / 1000000000000000000000000000n * 500000000000000000000000000n / 1000000000000000000000000000n = 250000000000000000750000000000000000000000000n
```

there you go. now we have salvaged the `75` right? We will proceed with this value for now.

Now, in get_borrow_amount:

```
// 2400n * (10n**6n) * 250000000000000000750000000000000000000000000n = 600000000000000001800000000000000000000000000000000000n
let amount_base_token = token_price.into() * borrow_capacity;
// 600000000000000001800000000000000000000000000000000000n / (10n**18n) = 600000000000000001800000000000000000n
let amount_quote_token = amount_base_token / decimals_difference.into();
// (600000000000000001800000000000000000n - 0n) / 100n * 98n = 588000000000000001764000000000000000n = 588.000000000000001764 USDC
((amount_quote_token - total_borrowed) / 100_u256 * borrow_const.into())
```

So we are left with 588.000000000000001764 USDC. We know that USDC's decimal places is 6 digits. Should we then care about 0.000000000000001764 USDC? No. Because there is absolutely no way to describe this value in USDC. So it's totally fine to ignore this value. This means we won't have to scale the borrow capacity by `ZK_SCALE_DECIMALS`.

But what about other stablecoins like DAI? DAI has 18 decimal places. Let's check if doing so would make some difference for DAI. 

First, the price would be `2400n * (10n**18n)` because DAI has 18 decimals. We will assume the same scaled version of borrowing capacity, which is `borrow capacity = 250000000000000000750000000000000000000000000n`. 

Then, get_borrow_amount would go like

```
// 2400n * (10n**18n) * 250000000000000000750000000000000000000000000n = 600000000000000001800000000000000000000000000000000000000000000000n
let amount_base_token = token_price.into() * borrow_capacity;
// 600000000000000001800000000000000000000000000000000000000000000000n / (10n**18n) = 600000000000000001800000000000000000000000000000n
let amount_quote_token = amount_base_token / decimals_difference.into();
// (600000000000000001800000000000000000000000000000n - 0n) / 100n * 98n = 588000000000000001764000000000000000000000000000n = 588.000000000000001764 DAI
((amount_quote_token - total_borrowed) / 100_u256 * borrow_const.into())
```

Is 588.000000000000001764 DAI a valid value in DAI? Absolutely, because it supports up to 18 decimal places. What if you didn't scale the borrow capacity by ZK_SCALE_DECIMALS? Well, `borrow_capacity` would have been `250000000000000000n` and we would lose certain amount of DAI here. For the sake of completeness, let's do the calculation again:

```
// 2400n * (10n**18n) * 250000000000000000n = 600000000000000000000000000000000000000n
let amount_base_token = token_price.into() * borrow_capacity;
// 600000000000000000000000000000000000000n / (10n**18n) = 600000000000000000000n
let amount_quote_token = amount_base_token / decimals_difference.into();
// (600000000000000000000n - 0n) / 100n * 98n = 588000000000000000000n = 588.000000000000000000n DAI
((amount_quote_token - total_borrowed) / 100_u256 * borrow_const.into())
```

So if you just pass borrowing_capacity without scaling it, you will end up losing the last few digits. So anyway, my recommendation would be to scale the borrow capacity by some big number, like `ZK_SCALE_DECIMALS`. Then all will be good. The same principle would apply to any other divisions.